EIPS/eip-3540.md

```diff
@@ -79,58 +81,91 @@ The container starts with the EOF header:
-|--------------|------------------------------------------------|
-| 0            | *reserved for section headers terminator byte* |
-| 1            | code                                           |
-| 2            | data                                           |
+##### Body
+
+| name             | length   | value         | description |
+|------------------|----------|---------------|-------------|
+| type_section     | variable | n/a       | stores EIP-4750 and EIP-5450 code section metadata  |
+| inputs           | 1 byte   | 0x00-0x7F | number of stack elements the code section consumes  |
+| outputs          | 1 byte   | 0x00-0x7F | number of stack elements the code section returns   |
+| max_stack_height | 2 bytes  | 0x00-0x3FF | max height of data stack during execution           |
+| code_section     | variable | n/a       | arbitrary bytecode                                  |
+| data_section     | variable | n/a       | arbitrary sequence of bytes                         |
+
+See [EIP-4750](./eip-4750.md) for more information on the type section content.
 
 #### EOF version 1 validation rules
 
 1. In addition to general validation rules above, EOF version 1 bytecode conforms to the rules specified below:
-   - Exactly one code section MUST be present.
-   - The code section MUST be the first section.
-   - A single data section MAY follow the code section.
-2. Any other version is invalid.
+   - One type section header MUST be present immediately following the EOF version. Each code section MUST have a specified type signature in the type body.
+   - One code section header MUST be present immediately following the type section. A maxmimum of 1024 individual code sections are allowed.
+   - One data section header MUST be present immediately following the code section.
+2. Any version other than `0x01` is invalid.
 
 (*Remark:* Contract creation code SHOULD set the section size of the data section so that the constructor arguments fit it.)
 
 ### Changes to execution semantics
 
 For clarity, the *container* refers to the complete account code, while *code* refers to the contents of the code section only.
 
-1. `JUMPDEST`-analysis is only run on the *code*.
-2. Execution starts at the first byte of the *code*, and `PC` is set to 0.
-3. Execution stops if `PC` goes outside the code section bounds.
-4. `PC` returns the current position within the *code*.
-5. `JUMP`/`JUMPI` uses an absolute offset within the *code*.
-6. `CODECOPY`/`CODESIZE`/`EXTCODECOPY`/`EXTCODESIZE`/`EXTCODEHASH` keeps operating on the entire *container*.
-7. The input to `CREATE`/`CREATE2` is still the entire *container*.
-8. The size limit for deployed code as specified in [EIP-170](./eip-170.md) and for init code as specified in [EIP-3860](./eip-3860.md) is applied to the entire *container* size, not to the *code* size.
+1. Execution starts at the first byte of the first code section, and PC is set to 0.
```

Comment:
If we are removing the `PC` operation does it really matter what we call the first byte of the code section?